collated/main/jelneo.md

```diff
@@ -0,0 +1,1244 @@
+        } catch (IllegalValueException ive) {
+            assert false : "New debt amount should not be invalid since amount and debt field in target have "
+                    + "been validated";
+        }
+
+        return persons.getReadOnlyPerson(index);
+    }
+
+```
+###### \java\seedu\address\model\ModelManager.java
+``` java
+    /**
+     * Authenticates user
+     * @throws UserNotFoundException if username and password does not match those in the user preference file
+     * @throws IllegalValueException if username and password does not meet username and password requirements
+     */
+    public void authenticateUser(Username username, Password password) throws UserNotFoundException,
+            IllegalValueException {
+        Username fileUsername = new Username(getUsernameFromUserPref());
+        if (fileUsername.equals(username) && checkAgainstPasswordFromUserPref(password.toString(),
+                userPrefs.getPasswordSalt())) {
+            raise(new LoginAppRequestEvent(true));
+        } else {
+            raise(new LoginAppRequestEvent(false));
+            throw new UserNotFoundException();
+        }
+    }
+
+    /**
+     * Logs user out
+     */
+    public void logout() {
+        raise(new LoginAppRequestEvent(false));
+    }
+
+    public String getUsernameFromUserPref() {
+        return userPrefs.getAdminUsername();
+    }
+
+    public String getPasswordFromUserPref() {
+        return userPrefs.getAdminPassword();
+    }
+
+    /**
+     * Checks the entered password against the password stored in {@code UserPrefs} class
+     * @param currPassword password entered by user
+     * @return true if the hash generated from the entered password matches the hashed password stored
+     * in {@code UserPrefs}
+     */
+    public boolean checkAgainstPasswordFromUserPref(String currPassword, byte[] salt) {
+        String hashedPassword = getSha512SecurePassword(currPassword, salt);
+        return hashedPassword.equals(userPrefs.getAdminPassword());
+    }
+
+    /**
+     * Increase the debt of a person by the amount indicated
+     * @param target person in the address book who borrowed more money
+     * @param amount amount that the person borrowed. Must be either a positive integer or positive number with
+     *               two decimal places
+     * @throws PersonNotFoundException if {@code target} could not be found in the list.
+     */
+    @Override
+    public void addDebtToPerson(ReadOnlyPerson target, Debt amount) throws PersonNotFoundException {
+        addressBook.addDebtToPerson(target, amount);
+        indicateAddressBookChanged();
+    }
+
+    /**
+     * Decrease the debt of a person by the amount indicated
+     * @param target person in the address book who paid back some money
+     * @param amount amount that the person paid back. Must be either a positive integer or positive number with
+     *               two decimal places
+     * @return repayingPerson
```

Comment:
Do not leave javadoc tags without descriptions.
If there is no need for the description, it is fine to remove the tag.